modules/core/src/controllers/map-controller.js

```diff
@@ -104,6 +106,8 @@ class MapState extends ViewState {
       startPitch,
       startZoom
     };
+
+    this.ViewportType = ViewportType;
```

Comment:
This does not feel great, but I get why. 

I wonder if it could be ViewType instead, so we could use the View.makeViewport method and be less dependent on the named Viewport classes that I have been wanting to get rid of. Having MapState know about MapView feels slightly more OK than having it know about WebMercatorViewport...

Reply:
This pattern is also used by OrbitController/OrthographicController. I will address this in a follow-up PR.